### PR TITLE
fix how we find parent process

### PR DESCRIPTION
**Pull request by cfbao**
### Why

Turns out BusyBox only implements a small portion of the [POSIX spec][1] for its [ps][2], so our code that gets the parent process (and hence `bmx print`) never worked on Alpine...
(I guess no one ever used `bmx print` on alpine...)

Switch to use native API call, which is actually easier.

[1]: https://pubs.opengroup.org/onlinepubs/9699919799/utilities/ps.html
[2]: https://boxmatrix.info/wiki/Property:ps

### Ticket

https://desire2learn.atlassian.net/browse/VUL-403

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -45,9 +45,9 @@ jobs:
             Throw "Tag name format incorrect"
           }
           if ($env:PRERELEASE -eq 'true') {
-              $releaseUrl = gh release create "$env:RELEASE_TAG" --repo Brightspace/bmx --draft --prerelease --generate-notes --title "Release $env:RELEASE_TAG"
+              $releaseUrl = gh release create "$env:RELEASE_TAG" --repo Brightspace/bmx --draft --prerelease --generate-notes --title "Release $env:RELEASE_TAG" --target $env:GITHUB_SHA
           } else {
-              $releaseUrl = gh release create "$env:RELEASE_TAG" --repo Brightspace/bmx --draft --generate-notes --title "Release $env:RELEASE_TAG"
+              $releaseUrl = gh release create "$env:RELEASE_TAG" --repo Brightspace/bmx --draft --generate-notes --title "Release $env:RELEASE_TAG" --target $env:GITHUB_SHA
```
Comment (cfbao) [inline .github/workflows/publish.yml:50]: unrelated fix - our release job only worked on the main branch.

File: src/D2L.Bmx/ParentProcess.cs [hunk 2]
```diff
@@ -2,43 +2,41 @@
...
 /**
 This file may eventually be replaced if the following dotnet runtime api proposals are implemented:
 - https://github.com/dotnet/runtime/issues/21941
 - https://github.com/dotnet/runtime/issues/24423
-
-Uses the same approach of calling NtQueryInformationProcess as in the PowerShell library
-https://github.com/PowerShell/PowerShell/blob/26f621952910e33840efb0c539fbef1e2a467a0d/src/System.Management.Automation/engine/ProcessCodeMethods.cs
 */
-internal partial class WindowsParentProcess {
+internal partial class ParentProcess {
```
Comment (cfbao) [inline src/D2L.Bmx/ParentProcess.cs:11]: Consolidated both the previous Windows and Unix classes under the same one now.
The high level method `GetParentProcessName` has identical logic.

File: src/D2L.Bmx/ParentProcess.cs [hunk 3]
```diff
@@ -2,43 +2,41 @@
...
+	public static string? GetParentProcessName() {
+		int parentPid = -1;
+		if( OperatingSystem.IsWindows() ) {
+			parentPid = GetWindowsParentProcessId();
+		} else if( OperatingSystem.IsLinux() || OperatingSystem.IsMacOS() ) {
+			parentPid = GetPosixParentProcessId();
+		}
+		if( parentPid == -1 ) {
+			return null;
+		}
```
Comment (cfbao) [inline src/D2L.Bmx/ParentProcess.cs:21]: This is new.
`GetWindowsParentProcessId` returns -1 on error. We should check that, instead of passing it to `GetProcessById`, which will throw.

File: src/D2L.Bmx/ParentProcess.cs [hunk 4]
```diff
@@ -2,43 +2,41 @@
...
+		return Process.GetProcessById( parentPid ).ProcessName;
+	}
 
-	[LibraryImport( "ntdll.dll", EntryPoint = "NtQueryInformationProcess" )]
-	internal static partial int NtQueryInformationProcess(
```
Comment (cfbao) [inline src/D2L.Bmx/ParentProcess.cs:16]: Moved below, as per our convention to put public/high-level methods first, and private/low-level/helper methods after.
Also changed it to `private`, as it's not used elsewhere.

**Review by gord5500 — APPROVED** [review]
